### ISSUE TITLE
How to add a bug project directly to the Maui code repository and build this bug project so you can test or find a solution inside the Maui code?

### ISSUE THREAD

**Issue by jonmdev** [dotnet/maui#21201]
### Description

I have cloned a copy of the Maui repository via GitHub which then looks like this when I open it in Visual Studio:

<img width="696" alt="maui repo" src="https://github.com/dotnet/maui/assets/43948450/904deee1-f284-4219-8075-ef47366cb83e">

I would like to try to fix some problems as I am very far into Maui and it does everything I need minus only a few bugs.

However, I cannot figure out how I can use the main Maui repository/project to test or fix a problem.

## Case Study

I have this bug report which shows Maui/Android is clipping and wrapping Labels abnormally: https://github.com/dotnet/maui/issues/17884

I suspect it has to do with this other bug report I also made which shows that native Android text measurements do not match Maui's Label.Measure() method: https://github.com/dotnet/maui/issues/18328

My theory is that if Label.Measure() does not give the same results as Android's native text measurement system, this could cause the clipping or wrapping.

### Next Step?
I think the next detective step (unless there is another suggestion) would be to take the label measurement bug code and make a project inside the Maui repository with it. Then when I run that code, I can debug out data from the measurement process through the Maui code and see perhaps why it is discordant.

I don't know. How do I do anything like this? Step by step, how do I add a separate project inside the Maui project and then run it? Or what are we supposed to do to figure out and solve these problems?

### Maui Can't Access Everything?

Complicating this perhaps: I tried stepping through the Maui code and as I can see it, Label.Measure() works like this:
-  Label.Measure() is inside VisualElement.cs: calls VisualElement.Measure() which inside calls VisualElement.GetSizeRequest() which calls VisualElement.OnMeasure() which calls Handler.GetDesiredSize(), which comes from the IViewHandler Interface
- Label's handler is of type Microsoft.Maui.Handlers.LabelHandler, which I believe is derived from ViewHandler
- ViewHandler in ViewHandlerOfT.Android.cs contains the GetDesiredSize() function which calls this.GetDesiredSizeFromHandler()
 - ... which then calls: PlatformInterop.MeasureAndGetWidthAndHeight() but this shows as not existing???

In other words, I followed this as far as I could and then came to this:

<img width="601" alt="maui repo platforminterop" src="https://github.com/dotnet/maui/assets/43948450/e24ccb86-14de-4a55-9a57-5cbf08cb8d13">

What does that mean? How would I track the problem down further if it is a dead end here? Is this going into some other repo dependency like native Android libraries that are stored separately? If so how can it be figured out?

## Tl;Dr ...
I want to find the exact code that is being run to get the expected Label size in Android Maui's code for Layout (wherever the heck that is - can't find yet - any suggestions also?) and edit/debug it while running my bug projects to find/fix the problem. How do I do this?

Thanks for any help.

### Version with bug

8.0.3 GA

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

Android

**Comment by jonathanpeppers** [comment]
Platform interop is written in Java and setup so MAUI has an internal C# binding for it:

* https://github.com/dotnet/maui/blob/main/src/Core/AndroidNative/maui/src/main/java/com/microsoft/maui/PlatformInterop.java

This is the "slim binding" idea -- that phrase has been thrown around by MAUI PMs lately.
